docs/README.rst

```diff
@@ -162,39 +194,149 @@ Configuration Parameters
+- Cloud Foundry NATS message bus to subscribe to for routing information;
+- Cloud Foundry NATS message bus login credentials;
+- Cloud Foundry OAuth endpoint for API access;
+- Cloud Foundry OAuth API access credentials;
+- Cloud Foundry routing API endpoint;
+- Type of routes to watch in Cloud Foundry (all routes, in this case).
 
-Example Configuration Files
-```````````````````````````
+.. literalinclude:: /_static/config_examples/example-manifest.yml
 
-- :fonticon:`fa fa-download` :download:`sample-cf-manifest-structure.yml </_static/config_examples/manifest-structure.yml>`
-- :fonticon:`fa fa-download` :download:`sample-cf-config-structure.yml </_static/config_examples/config-structure.yml>`
 - :fonticon:`fa fa-download` :download:`sample-cf-app-manifest.yml </_static/config_examples/example-manifest.yml>`
+- :fonticon:`fa fa-download` :download:`sample-cf-config-structure.yml </_static/config_examples/config-structure.yml>`
+
+.. _route types:
+
+Route Types
+-----------
+
+The |cfctlr| accepts three (3) values for the ``route_type`` configuration parameter.
+Each of the supported ``route_mode`` options have different configuration requirements:
+
+- ``all``: Requires :ref:`nats <nats-configs>`, :ref:`routing_api <routing-api-configs>`, and :ref:`oauth <oauth-configs>`.
+- ``http``: Requires :ref:`nats <nats-configs>`.
+- ``tcp``: Requires :ref:`routing_api <routing-api-configs>` and :ref:`oauth <oauth-configs>`.
+
+When managing HTTP routes, the |cfctlr| will create an HTTP virtual server (virtual address port 80) for routing into `Cloud Foundry`_.
+If you define an SSL profile in the configuration (the ``ssl_profiles`` parameter), the Controller creates an additional HTTPS virtual server (virtual address port 443).
+You can attach multiple certificate/key pairs to the HTTPS virtual server.
+The BIG-IP device uses `TLS Server Name Indication`_ (SNI) to choose the correct certificate to present to the client; SNI allows the `Cloud Foundry`_ instance to support multiple hostnames (foo.mycf.com and bar.mycf.com).
+Some of these cert/key pairs can be wildcard (\*.mycf.com).
+
+.. _health checks:
+
+Health Checks
+-------------
+
+The |cfctlr| supports native Cloud Foundry health checks for itself and will manage BIG-IP health checks for applications.
+
+To configure Cloud Foundry health checks for the |cfctlr| (this ensures the orchestration system will manage the Controller availability for you), define the ``health-check-type`` and ``health-check-http-endpoint`` settings as follows:
+
+.. code-block:: yaml
+
+   health-check-type: http
+   health-check-http-endpoint: /health
+
+The |cfctlr| will also manage BIG-IP health checking of the managed applications. To use any health monitor(s) that already exists on the BIG-IP system, add the name to the application manifest under ``bigip.health_monitors``. As these monitors are applied to all applications in the system the |cfctlr| defaults to using the ``/Common/tcp_half_open`` monitor.
```

Comment:
"Are applied" fails the active/passive grammar test. Change to:

"Because these monitors apply to all applications in the system, the |cfctlr| uses the ``/Common/tcp_half_open`` monitor by default."